upload/src/addons/SV/TitleEditHistory/XF/Entity/Thread.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace SV\TitleEditHistory\XF\Entity;
+
+use XF\Mvc\Entity\Structure;
+
+class Thread extends XFCP_Thread {
+
+	public function canViewThreadTitleHistory(&$error = null) {
+		if (! \XF::visitor()->user_id)
+		{
+			return false;
+		}
+
+		if (! \XF::options()->editHistory['enabled'])
+		{
+			return false;
+		}
+
+		if (\XF::visitor()->hasNodePermission($this->node_id, 'manageAnyThread'))
+		{
+			return true;
+		}
+
+		return false;
+	}
+
+	protected function _isNotInsideSilentEditWindow() {

```

Comment:
The edit history logic should be moved into `XF\Service\Thread\Editor` similar to `XF\Service\Post\Editor::setMessage`, `XF\Service\Post\Editor::setMessage::setupEditHistory`.